src/Nethermind/Nethermind.Db.Rocks/Statistics/DbMetricsUpdater.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Db.Rocks.Config;
+using Nethermind.Logging;
+using RocksDbSharp;
+
+namespace Nethermind.Db.Rocks.Statistics
+{
+    public class DbMetricsUpdater
+    {
+        private readonly string _dbName;
+        private readonly DbOptions _dbOptions;
+        private readonly RocksDb _db;
+        private readonly IDbConfig _dbConfig;
+        private readonly ILogger _logger;
+        private Timer? _timer;
+
+        public DbMetricsUpdater(string dbName, DbOptions dbOptions, RocksDb db, IDbConfig dbConfig, ILogger logger)
+        {
+            _dbName = dbName;
+            _dbOptions = dbOptions;
+            _db = db;
+            _dbConfig = dbConfig;
+            _logger = logger;
+        }
+
+        public void StartUpdating()
+        {
+            var offsetInSec = _dbConfig.StatsDumpPeriodSec * 1.1;
```

Comment:
Why 1.1?

Reply:
Just a number bigger than StatsDumpPeriodSec 
By default it's 10*1.1 = 11 minutes
So we extract stats after a dump has been done.